pkg/resolver/related.go

```diff
@@ -140,15 +142,28 @@ func (s *SearchResult) buildRelationsQuery() {
 		"related.level")
 	relQueryInnerJoin := relQuery.InnerJoin(goqu.S("search").Table("resources"),
 		goqu.On(goqu.Ex{"related.uid": goqu.L(`"resources".uid`)}))
-	relQueryWithRbac := relQueryInnerJoin.Where(buildRbacWhereClause(s.context, s.userData))
+	//RBAC CLAUSE
+	relQueryWithRbac := relQueryInnerJoin //without RBAC
+	withoutRBACSql, _, withoutRBACSqlErr := relQueryInnerJoin.ToSQL()
+	klog.V(5).Info("Relations query before RBAC:", withoutRBACSql, withoutRBACSqlErr)
+
+	if s.userData != nil && !Iskubeadmin(s.context) {
+		// add rbac
+		relQueryWithRbac = relQueryInnerJoin.Where(buildRbacWhereClause(s.context, s.userData))
+	} else {
+		if !Iskubeadmin(s.context) {
+			panic(fmt.Sprintf("RBAC clause is required! None found for search relations query %+v for user %s ", s.input,
+				s.context.Value(rbac.ContextAuthTokenKey)))
+		}
```

Comment:
Please log a warning to remind us to remove this workaround.
```suggestion
		} else {
			klog.warning("TEMPORARY WORKAROUND: Skipping rbac for cluster admin")
		}
```

Reply:
We have it here https://github.com/stolostron/search-v2-api/blob/1d18c1d387b6468f628b5b4e67d1cd64bf1bf5d2/pkg/resolver/search.go#L130. Should we add it specifically for this function?

Reply:
ok, the existing warning should be enough.  
I'm just worried we don't forget to rollback anything.